infrastructure/sentry-subgraph/src/referee.ts

```diff
@@ -669,3 +672,36 @@ export function handleBulkRewardsClaimed(event: BulkRewardsClaimedEvent): void {
   }
 
 }
+
+export function handleNodeConfirmed(event: NodeConfirmedEvent): void {
+  //subgraph entity id is nodeNum since its already unique
+  const nodeConfirmedEvent = new NodeConfirmation(
+    event.params.nodeNum.toString()
+  )
+  nodeConfirmedEvent.nodeNum = event.params.nodeNum;
+  nodeConfirmedEvent.blockHash = event.params.blockHash;
+  nodeConfirmedEvent.sendRoot = event.params.sendRoot;
+
+  //constructing confirmHash exactly how its built in the smart contract (RollupLib.confirmHash)
+  let blockHashBytes = nodeConfirmedEvent.blockHash;
+  let sendRootBytes = nodeConfirmedEvent.sendRoot;
+
+  if (!blockHashBytes || !sendRootBytes) {
+    log.warning("Failed to assign null event value to variable", []);
+    return;
+  }
+
+  //confirmData = keccak256(blockHash + sendRoot)
+  let concatenatedHexStr = blockHashBytes.concat(sendRootBytes).toHexString();
+  let concatenatedByteArray = ByteArray.fromHexString(concatenatedHexStr);
+  let confirmHashByteArray = crypto.keccak256(concatenatedByteArray);
+  let confirmHashHexStr = confirmHashByteArray.toHexString();
```

Comment:
Thats some crypto, thats very cool, it will sync much faster than calling the smart contract. Just keep in mind code complexity vs sync time when calling the smart contract struct.

Reply:
Agreed, since the logic here is very simple I opted to do it in the handler to save the contract hits 👍